hail/python/hail/ir/ir.py

```diff
@@ -2077,14 +2077,11 @@ def renderable_bindings(self, i, default_value=None):
 class StreamAgg(IR):
     @typecheck_method(a=IR, value_name=str, body=IR)
     def __init__(self, a, value_name, body):
-        a = a.handle_randomness(body.uses_randomness)
-        if body.uses_randomness:
+        a = a.handle_randomness(body.uses_agg_randomness)
+        if body.uses_agg_randomness:
             tup, uid, elt = unpack_uid(a.typ)
-            body = AggLet(value_name, elt, body)
-            if body.uses_value_randomness:
-                body = Let('__rng_state', RNGStateLiteral(), body)
-            if body.uses_agg_randomness(is_scan=False):
-                body = with_split_rng_state(body, is_scan=False)
+            body = AggLet(value_name, elt, body, is_scan=False)
+            body = with_split_rng_state(body, uid, is_scan=False)
```

Comment:
The test caught some problems with this. Most substantially, we shouldn't be making a new root `RNGStateLiteral` for any value randomness. That means if this occurs inside some other looping context, value randomness will be identical on each outer iteration. Instead, we should just use the rng state bound in the outer context, which means we don't need to do anything here.

Reply:
Thanks for explaining. Clears up why this was necessary.